src/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -273,41 +272,64 @@ public void loadVCFVersion() {
      */
     private void loadMetaDataMaps() {
         for (final VCFHeaderLine line : mMetaData) {
-            if ( line instanceof VCFInfoHeaderLine )  {
-                final VCFInfoHeaderLine infoLine = (VCFInfoHeaderLine)line;
-                addMetaDataMapBinding(mInfoMetaData, infoLine);
-            } else if ( line instanceof VCFFormatHeaderLine ) {
-                final VCFFormatHeaderLine formatLine = (VCFFormatHeaderLine)line;
-                addMetaDataMapBinding(mFormatMetaData, formatLine);
-            } else if ( line instanceof VCFFilterHeaderLine ) {
-                final VCFFilterHeaderLine filterLine = (VCFFilterHeaderLine)line;
-                mFilterMetaData.put(filterLine.getID(), filterLine);
-            } else if ( line instanceof VCFContigHeaderLine ) {
-                contigMetaData.add((VCFContigHeaderLine)line);
-            } else {
-                mOtherMetaData.put(line.getKey(), line);
+            addMetadataHeaderLine(line);
+        }
+    }
+
+    /**
+     * add a single header line to the appropriate map/list depending on its type
```

Comment:
Mention in docs here that lines with pre-existing keys will be discarded and not added to the header